### PR TITLE
RBAC: Add userLogin filter to the permission search endpoint

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

This PR adds a new query parameter to the search users permissions endpoint, to filter based on `userLogin`.

**Why do we need this feature?**

Plugin backends do not have access to the user `id` (see backend.proto of the plugin-sdk), but they do have access to the user `login`.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/registry/apis/datasource/standalone_services.go [hunk 1]
```diff
@@ -73,11 +76,23 @@ func apiBuilderServices(cfg *setting.Cfg, pluginID string) (
...
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+	teamService := teamimpl.ProvideService(sqlStore, cfg)
+	userService, err := userimpl.ProvideService(sqlStore, orgService, cfg, teamService, cacheService, quota, bundleRegistry)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	acimplService, err := acimpl.ProvideService(cfg, sqlStore, routeRegisterImpl, cacheService, accessControl, userService, featureToggles)
```
Comment (gamab) [inline pkg/registry/apis/datasource/standalone_services.go:92]: I did not expect that we had places in the code where wire isn't used

Comment (alexanderzobnin) [reply pkg/registry/apis/datasource/standalone_services.go:92]: yeah, looks a bit strange

File: pkg/services/accesscontrol/api/api.go [hunk 2]
```diff
@@ -101,7 +115,7 @@ func (api *AccessControlAPI) searchUserPermissions(c *contextmodel.ReqContext) r
 	userIDString := web.Params(c.Req)[":userID"]
 	userID, err := strconv.ParseInt(userIDString, 10, 64)
 	if err != nil {
-		response.Error(http.StatusBadRequest, "user ID is invalid", err)
+		return response.Error(http.StatusBadRequest, "user ID is invalid", err)
```
Comment (Jguer) [inline pkg/services/accesscontrol/api/api.go:118]: :flushed:

**Review by Jguer — APPROVED** [review]

File: pkg/services/accesscontrol/acimpl/service.go [hunk 3]
```diff
@@ -243,8 +245,25 @@ func (s *Service) DeclarePluginRoles(ctx context.Context, ID, name string, regs
...
+		// Reroute to the user specific implementation of search permissions
+		// because it leverages the user permission cache.
+		userPerms, err := s.SearchUserPermissions(ctx, usr.GetOrgID(), options)
+		if err != nil {
+			return nil, err
+		}
+		return map[int64][]accesscontrol.Permission{options.UserID: userPerms}, nil
+	}
+
 	timer := prometheus.NewTimer(metrics.MAccessSearchPermissionsSummary)
```
Comment (alexanderzobnin) [inline pkg/services/accesscontrol/acimpl/service.go:267]: Seems like observation happens after first return, so in some cases we do not measure it.

Comment (gamab) [reply pkg/services/accesscontrol/acimpl/service.go:267]: Yes, that's intended. The `SearchUserPermission` function has another metric.

File: pkg/services/accesscontrol/api/api.go (hunk 2 above)
Comment (gamab) [reply pkg/services/accesscontrol/api/api.go:118]: Yeah :sweat_smile: 
I just noticed there is the same a few lines below. But I'll tackle this in another PR (by removing the endpoint altogether :grimacing: )

Comment (IevaVasiljeva) [reply pkg/services/accesscontrol/api/api.go:118]: Oh nooo 🙈

**Review by IevaVasiljeva — APPROVED** [review]

**Pull request merged**